File: S5/HLIN505 - Modelisation et Programmation par Objets 2/TP/2016-1017/TP3/Pokemon/src/PokemonCasanier.java
public class PokemonCasanier extends PokemonPhysique {
	private int nbHeuresDevantLaTele;
	public PokemonCasanier(String name , double poids , int nombreDePattes , double taille , int nbHeuresDevantLaTele) {
		super(name , poids , nombreDePattes , taille);
		this.nbHeuresDevantLaTele = nbHeuresDevantLaTele;
	}
	public String toString() {
		return super.toString() + " et je passe " + nbHeuresDevantLaTele + " heures devant la tele par jour";
	}
}

File: S6/HLIN603 - Objets avances/TPs/TP1/Exercice 2/source/PokemonCasanier.cc
#include "../header/PokemonCasanier.h"
#include <sstream>

PokemonCasanier::PokemonCasanier(std::string s , int n , int m , float t , int h)
: PokemonTerrestre(s , n , m , t) , heuresDevantLaTeleParJour(h) {}

std::string PokemonCasanier::toString() {

  std::stringstream s;
  std::string sHeuresDevantLaTeleParJour;
  s << this->heuresDevantLaTeleParJour;
  s >> sHeuresDevantLaTeleParJour;

  return PokemonTerrestre::toString() + " et je passe " + sHeuresDevantLaTeleParJour + " heures devant la tele chaque jour";
}
